openthread/CMakeLists.txt

```diff
@@ -3,38 +3,39 @@
 #
 # SPDX-License-Identifier: LicenseRef-BSD-5-Clause-Nordic
 #
-include(${NRFXLIB_DIR}/common.cmake)
-nrfxlib_calculate_lib_path(lib_path)
+include(cmake/extensions.cmake)
 
-include(${CMAKE_CURRENT_LIST_DIR}/cmake/extensions.cmake)
-
-function(checkOpenthreadVersion)
-  file(READ "${ZEPHYR_OPENTHREAD_MODULE_DIR}/../include/openthread/instance.h" INSTANCE_FILE_CONTENT)
-  string(REGEX MATCH "#define[ ]*OPENTHREAD_API_VERSION[ ]*\\(([0-9]+)\\)" MACRO_DEF "${INSTANCE_FILE_CONTENT}")
-  set(OPENTHREAD_SOURCE_API_VERSION ${CMAKE_MATCH_1})
-
-  file(READ "include/openthread/instance.h" INSTANCE_FILE_CONTENT)
-  string(REGEX MATCH "#define[ ]*OPENTHREAD_API_VERSION[ ]*\\(([0-9]+)\\)" MACRO_DEF "${INSTANCE_FILE_CONTENT}")
-  set(OPENTHREAD_LIB_API_VERSION ${CMAKE_MATCH_1})
-
-  #check if the Openthread API changed
-  if(NOT (${OPENTHREAD_SOURCE_API_VERSION} EQUAL ${OPENTHREAD_LIB_API_VERSION}))
-    message(WARNING "The Openthread source code version (${OPENTHREAD_SOURCE_API_VERSION}) is not equal to compiled Openthread library version (${OPENTHREAD_LIB_API_VERSION}).")
-  endif()
-endfunction()
+set(OPENTHREAD_LIB_MBEDTLS_CONF_FILE "nrf_security_mbedtls_configuration.h")
 
 if(CONFIG_OPENTHREAD_SOURCES)
   message(DEBUG "Building OT from sources, config file will be generated.")
 
-  set(OPENTHREAD_CONFIG_FILE "${CMAKE_BINARY_DIR}/openthread_lib_configuration.txt")
+  set(OPENTHREAD_CONFIG_FILE "${CMAKE_CURRENT_BINARY_DIR}/openthread_lib_configuration.txt")
 
   openthread_libs_configuration_write(${OPENTHREAD_CONFIG_FILE})
 
-  nrfxlib_calculate_lib_path(lib_path)
-  set(OPENTHREAD_DST_DIR "${CMAKE_CURRENT_LIST_DIR}/${lib_path}/v${CONFIG_OPENTHREAD_THREAD_VERSION}")
+  openthread_calculate_lib_path("v${CONFIG_OPENTHREAD_THREAD_VERSION}" OPENTHREAD_DST_DIR)
+
   set(OPENTHREAD_HEADERS_DIR "${ZEPHYR_OPENTHREAD_MODULE_DIR}/../include")
+  set(OPENTHREAD_HEADERS_DST_DIR "${NRFXLIB_DIR}/openthread/include")
+  set(OPENTHREAD_SYSTEM_H "${ZEPHYR_OPENTHREAD_MODULE_DIR}/../examples/platforms/openthread-system.h")
+  set(NRF_SECURITY_MBEDTLS_CONFIG_FILE ${CMAKE_CURRENT_BINARY_DIR}/../nrf_security/include/${CONFIG_MBEDTLS_CFG_FILE})
+
+  add_custom_target(strip_openthread_libraries
+    COMMAND ${CMAKE_COMMAND} -E echo "Stripping OpenThread libraries"
+    COMMAND ${CMAKE_STRIP} --strip-debug
+      $<TARGET_FILE:openthread-ftd>
+      $<TARGET_FILE:openthread-mtd>
+      $<TARGET_FILE:openthread-radio>
+      $<TARGET_FILE:openthread-cli-ftd>
+      $<TARGET_FILE:openthread-cli-mtd>
+      $<TARGET_FILE:openthread-ncp-ftd>
+      $<TARGET_FILE:openthread-ncp-mtd>
+      $<TARGET_FILE:openthread-rcp>
+  )
```

Comment:
This is very GNU centric.
Zephyr is working towards supporting more toolchains, so I think we should use the bintools abstraction here:
https://github.com/zephyrproject-rtos/zephyr/blob/master/cmake/bintools/bintools_template.cmake#L150-L158
https://github.com/zephyrproject-rtos/zephyr/blob/master/cmake/bintools/gnu/target_bintools.cmake#L85

For inspiration, see:
https://github.com/zephyrproject-rtos/zephyr/blob/master/CMakeLists.txt#L1221-L1226

Then this will also work when Zephyr starts to have support for more toolchains.

Note: consider to have one command per target_file, as not all strip tools may interpret a missing `outfile` as overriding the input file.

Reply:
Unfortunately bintools abstraction is provided by zephyr v2.4.0 and currently is not available in downstream.

Reply:
Done in accordance with your recommendation, but waits for update zephyr in current https://github.com/nrfconnect/sdk-nrf/pull/2918.

Reply:
> Unfortunately bintools abstraction is provided by zephyr v2.4.0 and currently is not available in downstream.

right, had forgotten that :)
Thanks for updating to Zephyr 2.4 :+1: